olive/passes/onnx/genai_model_exporter.py

```diff
@@ -69,25 +81,37 @@ def is_accelerator_agnostic(accelerator_spec: AcceleratorSpec) -> bool:
         return False
 
     def _run_for_config(
-        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+        self,
+        model: Union[PyTorchModelHandler, ONNXModelHandler],
+        data_root: str,
+        config: Dict[str, Any],
+        output_model_path: str,
     ) -> ONNXModelHandler:
         from onnxruntime_genai.models.builder import create_model
 
-        if not model.hf_config:
+        precision = config["precision"]
+        metadata_only = config["metadata_only"]
+
```

Comment:
could you add a check here that the handler type is pytorch when metadata_only=false and onnx when true? 

Otherwise, there are potential missing attribute errors like the second condition at line 95.

Reply:
Attributes have defaults so shouldn't they still be valid in either context?

Reply:
.hf_config is only an attribute for pytorch model handler. It will raise attribute error if you called it on an onnx handler

Reply:
Done!